### PR TITLE
Feat: Add strategy selection to dependabot automerge workflow

### PR DESCRIPTION
**Pull request by ingvaar**
### Type of Change

- [ ] Bugfix
- [x] Enhancement / new feature
- [ ] Refactoring
- [ ] Documentation

### Description

This PR adds a `strategy` option to dependabot automerge workflow.
It allows to select the merge strategy. The possible options are:
- `rebase`
- `merge`
- `squash` (default)

### Checklist

- [ ] Add relevant labels (for example type of change or patch/minor/major)
- [x] Make sure not to introduce some mistakes
- [x] Update documentation
- [x] Review the [Contributing Guideline](../blob/main/CONTRIBUTING.md) and sign CLA
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by ingvaar** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by flaxel** [comment]
The PR looks already good but would be nice to fix the linting action. 🙌

File: .github/workflows/template_automerge_dependabot.yml [hunk 1]
```diff
@@ -44,11 +48,20 @@ jobs:
...
-          if [ ${{ inputs.force }} == 'true' ]; then
-            gh pr merge "$PR_URL" --squash --admin
+          MERGE_COMMAND="gh pr merge \"$PR_URL\""
+
+          if [ "${{ inputs.force }}" == 'true' ] && [ "${{ inputs.rebase }}" == 'true' ]; then
+            MERGE_COMMAND+=" --rebase --admin"
+          elif [ "${{ inputs.force }}" == 'true' ]; then
+            MERGE_COMMAND+=" --squash --admin"
+          elif [ "${{ inputs.rebase }}" == 'true' ]; then
+            MERGE_COMMAND+=" --rebase"
```
Comment (soemo) [inline .github/workflows/template_automerge_dependabot.yml:58]:
```suggestion
          MERGE_COMMAND="gh pr merge \"$PR_URL\""

          if [ "${{ inputs.rebase }}" == 'true' ]; then
            MERGE_COMMAND+=" --rebase"
          else
            MERGE_COMMAND+=" --squash"
          fi
          if [ ${{ inputs.force }} == 'true' ]; then
            MERGE_COMMAND+=" --admin"
          fi  
```

**Review by soemo — CHANGES_REQUESTED** [review]
thanks for merging my first suggestion so fast. I wasn't even able to comment on it :-)

But I to discuss another point. 
Do we really need the `eval` command.


Can we not refactor it  in that way  
pseudo code
````
          MERGE_OPTIONS=''
          if [ "${{ inputs.rebase }}" == 'true' ]; then
            MERGE_OPTIONS+=" --rebase"
          else
            MERGE_OPTIONS+=" --squash"
          fi
          if [ ${{ inputs.force }} == 'true' ]; then
            MERGE_OPTIONS+=" --admin"
          fi
          else
            MERGE_OPTIONS+=" --auto --merge"
          fi
          echo "Executing merge command with the options: '${MERGE_OPTIONS}'"
          gh pr merge "$PR_URL" "${MERGE_OPTIONS}"
````

**Comment by ingvaar** [comment]
@soemo I really liked the concise approach you proposed 🙁 

We don't need the `eval` here, and the pseudo code you proposed is nice. 

But there is a concern here with the merge method:

```bash
if [ "${{ inputs.rebase }}" == 'true' ]; then
  MERGE_OPTIONS+=" --rebase"
else
  MERGE_OPTIONS+=" --squash"
fi

if [ "${{ inputs.force }}" == 'true' ]; then
  MERGE_OPTIONS+=" --admin"
else
  MERGE_OPTIONS+=" --auto --merge"
fi
```

With `rebase` as false and `force` as false, we will cumulate `--squash` and `--merge`.

Should I change the input to select the strategy instead of a boolean between `rebase` and `merge`/`squash`?

**Comment by soemo** [comment]
> Should I change the input to select the strategy instead of a boolean between `rebase` and `merge`/`squash`?

That is a nice idea 👍 to have a strategy as input. If we set squash as default, the new code will be get much easier

**Comment by flaxel** [comment]
Sounds great but it would be nice if it‘s backward compatible without a breaking change. 🙌

File: .github/workflows/template_automerge_dependabot.yml
Comment (flaxel) [inline .github/workflows/template_automerge_dependabot.yml:1]: Did you test the action change anywhere? 🤔

File: .github/workflows/template_automerge_dependabot.yml [hunk 2]
```diff
@@ -44,11 +48,29 @@ jobs:
...
-            gh pr merge "$PR_URL" --squash --admin
+            MERGE_COMMAND+=" --admin"
+          fi
           else
-            gh pr merge --auto --merge "$PR_URL"
+            MERGE_COMMAND+=" --auto --merge"
           fi
+
+          echo "Executing merge command: $MERGE_COMMAND"
+          eval $MERGE_COMMAND
```
Comment (flaxel) [inline .github/workflows/template_automerge_dependabot.yml:73]: I thought we want to remove the eval command?! 🤔

File: README.md [hunk 3]
```diff
@@ -47,6 +47,8 @@ jobs:
     with:
       # optional: ⚠️ only enable the force merge if you want to do the merge just now
       force: true
+      # optional: choose strategy when merging (default: squash)
+      strategy: rebase
```
Comment (flaxel) [inline README.md:51]: I would add all possible parameters to the documentation. 🙌 
```suggestion
      strategy: rebase, merge
```

**Review by flaxel — APPROVED** [review]
Really nice improvement! 🎉

**Review by timdittler — APPROVED** [review]
LGTM, but it would be great to test it in action before merging

**Review by wmentzel — CHANGES_REQUESTED** [review]
Hey @ingvaar !

The action ran into a problem in the lambda-edge-secure-media repo:

image

https://github.com/Staffbase/lambda-edge-secure-media/actions/runs/7867783102/job/21467013254?pr=91

**Comment by soemo** [comment]
How we move on with this PR?

**Comment by wmentzel** [comment]
> How we move on with this PR?

We need to test it after the last change.

**Comment by wmentzel** [comment]
> We need to test it after the last change.

We will test it in one of our repos with the next minor bump of a dependency. /cc @ingvaar

**Comment by wmentzel** [comment]
We again have to wait for suitable dependabot PR matching the condition: https://github.com/Staffbase/gha-workflows/blob/main/.github/workflows/template_automerge_dependabot.yml#L39

We'll run dependabot over the week to hopefully soon have some.

**Comment by wmentzel** [comment]
Good news: Finally, there was a dependbat PR to test this.
Bad news: It's not working. 😢 

image

https://github.com/Staffbase/media-analytics/actions/runs/8223143665/job/22485477812?pr=87

**Comment by wmentzel** [comment]
The most recent version (HEAD of this brach => d43d6f6daad3e4d70ab20d046d7a8b194f706fb1) of the action is not used in https://github.com/Staffbase/media-analytics/pull/87. Most likely due to caching...

image

**Comment by wmentzel** [comment]
Here a successful run: https://github.com/Staffbase/media-analytics/actions/runs/8224375573/job/22488121540?pr=87

This is the dependabot PR for testing: https://github.com/Staffbase/media-analytics/pull/87

@flaxel I think the force option never worked before 😅 : ff8cf9eb4f401c068353bc16b955524cf5c72c86

**Review by wmentzel — APPROVED** [review]

File: .github/workflows/template_automerge_dependabot.yml
Comment (wmentzel) [reply .github/workflows/template_automerge_dependabot.yml:1]: Yes: https://github.com/Staffbase/gha-workflows/pull/219#issuecomment-1987343924

File: .github/workflows/template_automerge_dependabot.yml [hunk 4]
```diff
@@ -44,11 +48,30 @@ jobs:
...
-            gh pr merge --auto --merge "$PR_URL"
+            MERGE_OPTIONS+=("--auto")
           fi
+
+          echo "Executing merge command with the options: '${MERGE_OPTIONS[*]}'"
+          gh pr merge "$PR_URL" "${MERGE_OPTIONS[@]}"
         env:
           PR_URL: ${{ github.event.pull_request.html_url }}
           GITHUB_TOKEN: ${{ steps.get_token.outputs.token }}
+          STRATEGY: ${{ inputs.strategy }}
```
Comment (flaxel) [inline .github/workflows/template_automerge_dependabot.yml:76]: What is the advantage to have it in the env vars? 🤔

**Comment by flaxel** [comment]
> @flaxel I think the force option never worked before 😅 : [ff8cf9e](https://github.com/Staffbase/gha-workflows/commit/ff8cf9eb4f401c068353bc16b955524cf5c72c86)

Why do you think that the action doesn't work? We are using it f.e. in the EXS and there we have no problems. 😄

**Review by flaxel — APPROVED** [review]

File: .github/workflows/template_automerge_dependabot.yml (hunk 4 above)
Comment (wmentzel) [reply .github/workflows/template_automerge_dependabot.yml]: There was a misunderstanding which Falk an I clarified in a call. 👍

**Comment by wmentzel** [comment]
> > @flaxel I think the force option never worked before 😅 : [ff8cf9e](https://github.com/Staffbase/gha-workflows/commit/ff8cf9eb4f401c068353bc16b955524cf5c72c86)
> 
> @wmentzel Why do you think that the action doesn't work? We are using it f.e. in the EXS and there we have no problems. 😄

I clarified this in a call with Falk. My assumption was based on a change which was introduced in this PR before I took over 😅. So, on main the action was never broken.

**Pull request merged**